HPCC-34163 code generator not included in coverage

AttilaVamos opened the pull request:
commonSetup.cmake:
- When the Platform build for coverage add definition of '_COVERAGE' into the environment (using later in eclcc.cpp)

eclcc.cpp:
- Use the '_COVERAGE' to replace '_exit(*)' with 'exit(*)'. The '_exit()' drops everything and prevent to call all clean-up and coverage related functions (like update .gcda files) and that causes nearly zero coverage.
- This change can cause test failure like this:
  637. Test: regex_arg_types.ecl eclccserver: (0,0): Error C9999: eclcc exited - could be because out of memory - see compile log for details

Tested in local coverage build and metrics generation.

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

ghalliday commented:
Change looks ok, but I am curious about the regex difference.

ghalliday commented on ecl/eclcc/eclcc.cpp:
```diff
@@ -639,7 +644,12 @@ int main(int argc, const char *argv[])
         //In release mode exit without calling all the clean up code.
         //It is faster, and it helps avoid potential crashes if there are active objects which depend on objects in file hook dlls.
         fflush(NULL);
+
```
A better fix may be to include this whole if() (lines 637..643) in a #ifndef _COVERAGE.

ghalliday reviewed (commented):
@AttilaVamos looks good.  One suggestion for a change.

ghalliday commented:
HPCC-34650 Created to cover the crash.

ghalliday reviewed (approved):
@AttilaVamos please squash

AttilaVamos committed:
HPCC-34163 code generator not included in coverage

commonSetup.cmake:
- When the Platform build for coverage add definition of '_COVERAGE' into the
  environment (using later in eclcc.cpp)

eclcc.cpp:
- Use the '_COVERAGE' to replace '_exit(*)' with 'exit(*)'.
  The '_exit()' drops everything and prevent to call all clean-up and coverage
  related functions (like update .gcda files) and that causes nearly zero
  coverage.
- This change can cause test failure like this:
  637. Test: regex_arg_types.ecl
  eclccserver: (0,0): Error C9999: eclcc exited - could be because out of
  memory - see compile log for details

Tested in local coverage build and metrics generation.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>
